### ISSUE TITLE
Investigate Santa Cruz Public Libraries `UnidentifiedImageError` issue, which crops up during the `content_harvesting` task

### ISSUE THREAD

**Issue by christinklez** [ucldc/rikolti#881]
### 27143
- Run ID: manual__2024-04-19T16:34:54+00:00
- Mapper: omeka.no_thumb
- Two errors during content harvesting step: page 33 & page 47 (error logs below)
   - [page 33](https://7a8067cb-3b99-477e-a883-7e311175a9b4.c3.us-west-2.airflow.amazonaws.com/dags/harvest_collection/grid?dag_run_id=manual__2024-04-19T16%3A34%3A54%2B00%3A00&task_id=content_harvesting.content_harvest&tab=mapped_tasks&num_runs=365&map_index=33):
      -  oai:history.santacruzpl.org:113478
      - is_shown_at: https://history.santacruzpl.org/omeka/items/show/113478
      - is_shown_by: https://history.santacruzpl.org/omeka/files/thumbnails/730dc7908dbb30f9e18ac336af09a413.jpg
      - on -test: https://calisphere-test.cdlib.org/item/058690c76101663e538e2b47938d2e51/
   - [page 47](https://7a8067cb-3b99-477e-a883-7e311175a9b4.c3.us-west-2.airflow.amazonaws.com/dags/harvest_collection/grid?dag_run_id=manual__2024-04-19T16%3A34%3A54%2B00%3A00&task_id=content_harvesting.content_harvest&tab=mapped_tasks&num_runs=365&map_index=47):
      - oai:history.santacruzpl.org:129307
      - is_shown_at: https://history.santacruzpl.org/omeka/items/show/129307
      - is_shown_by: https://history.santacruzpl.org/omeka/files/thumbnails/f98a6a0f011057e206ebecb30c5e3075.jpg
      - on -test: https://calisphere-test.cdlib.org/item/7485d02319dba4924e66040e9072d35d/

### 27148
- Run ID: manual__2024-04-19T18:49:44+00:00
- Mapper: omeka.no_thumb
- 2 errors during content harvesting step: page 0 & page 1 (error logs below)
   - [page 0](https://7a8067cb-3b99-477e-a883-7e311175a9b4.c3.us-west-2.airflow.amazonaws.com/dags/harvest_collection/grid?dag_run_id=manual__2024-04-19T18%3A49%3A44%2B00%3A00&num_runs=365&task_id=content_harvesting.content_harvest&tab=mapped_tasks&map_index=0) 
      -  oai:history.santacruzpl.org:86
      - is_shown_at: https://history.santacruzpl.org/omeka/items/show/86
      - is_shown_by: https://history.santacruzpl.org/omeka/files/thumbnails/a3b3a58e583a985aa6b65510210a8eb2.jpg
      - on -test: https://calisphere-test.cdlib.org/item/c7397e2c99b5aa6df3e94a1919702fea/
   - [page 1](https://7a8067cb-3b99-477e-a883-7e311175a9b4.c3.us-west-2.airflow.amazonaws.com/dags/harvest_collection/grid?dag_run_id=manual__2024-04-19T18%3A49%3A44%2B00%3A00&num_runs=365&task_id=content_harvesting.content_harvest&tab=mapped_tasks&map_index=1)
      -  oai:history.santacruzpl.org:109
      - is_shown_at: https://history.santacruzpl.org/omeka/items/show/109
      - is_shown_by: https://history.santacruzpl.org/omeka/files/thumbnails/d15d2283f6b2b3e7b7304fbe95bc30d0.jpg
      - on -test: https://calisphere-test.cdlib.org/item/4539ea9942b6e05b94463183013af184/

***

## 27143 content harvest page 33

**Content harvest error log:**
```
Harvesting content for 35 records at 27143/vernacular_metadata_2024-04-19T16:35:06/mapped_metadata_2024-04-19T16:57:54/data/1387.jsonl
[2024-04-19, 17:11:39 UTC] {{taskinstance.py:1345}} INFO - Marking task as FAILED. dag_id=harvest_collection, task_id=content_harvesting.content_harvest, map_index=33, execution_date=20240419T163454, start_date=20240419T170904, end_date=20240419T171139
[2024-04-19, 17:11:39 UTC] {{logging_mixin.py:150}} INFO - Message sent to SNS with Message ID: ef4c65a2-1b38-577e-b51e-3b3eb333e416
[2024-04-19, 17:11:39 UTC] {{standard_task_runner.py:104}} ERROR - Failed to execute job 76819 for task content_harvesting.content_harvest (This task is not in success state - last 100 logs from Cloudwatch:
Traceback (most recent call last):
  File "/content_harvester/by_record.py", line 150, in get_dimensions
    return Image.open(filepath).size
           ^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/PIL/Image.py", line 3339, in open
    raise UnidentifiedImageError(msg)
PIL.UnidentifiedImageError: cannot identify image file '/tmp/730dc7908dbb30f9e18ac336af09a413.jpg'
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/content_harvester/by_page.py", line 98, in <module>
    print_value.append(harvest_page_content(
                       ^^^^^^^^^^^^^^^^^^^^^
  File "/content_harvester/by_page.py", line 31, in harvest_page_content
    record_with_content = harvest_record_content(
                          ^^^^^^^^^^^^^^^^^^^^^^^
  File "/content_harvester/by_record.py", line 112, in harvest_record_content
    dimensions = get_dimensions(thumbnail.tmp_filepath, record['calisphere-id'])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/content_harvester/by_record.py", line 152, in get_dimensions
    raise Exception(
Exception: PIL.UnidentifiedImageError for calisphere-id oai:history.santacruzpl.org:113478: cannot identify image file '/tmp/730dc7908dbb30f9e18ac336af09a413.jpg'

```
**Mapped metadata:**
`{"calisphere-id": "oai:history.santacruzpl.org:113478", "collection_data": ["27143::Local News Index"], "collection_name": ["Local News Index"], "collection_url": ["27143"], "creator": ["Chesky, Laurel"], "date": ["2003-11-13"], "description": ["Salz Tannery"], "fetcher_type": ["oai"], "format": ["Text"], "id": "058690c76101663e538e2b47938d2e51", "identifier": ["CF-37608"], "is_shown_at": "https://history.santacruzpl.org/omeka/items/show/113478", "is_shown_by": "https://history.santacruzpl.org/omeka/files/thumbnails/730dc7908dbb30f9e18ac336af09a413.jpg", "item_count": 0, "language": ["English"], "mapper_type": ["oai.omeka.nothumb"], "media_source": {}, "publisher": ["Santa Cruz Public Libraries"], "repository_data": ["245::Santa Cruz Public Libraries"], "repository_name": ["Santa Cruz Public Libraries"], "repository_url": ["245"], "rights": ["Please contact the contributing institution for more information regarding the copyright status of this object."], "sort_collection_data": ["local news index::Local News Index::27143"], "sort_title": "trading spaces", "source": ["Good Times"], "subject": ["Tanneries", "Art and Artists"], "temporal": ["2000s"], "thumbnail_source": "https://history.santacruzpl.org/omeka/files/thumbnails/730dc7908dbb30f9e18ac336af09a413.jpg", "title": ["trading spaces"], "type": ["text"], "url_item": "https://history.santacruzpl.org/omeka/items/show/113478"}`

## 27143 content harvest page 47

**Content harvest error log:**
```
Harvesting content for 33 records at 27143/vernacular_metadata_2024-04-19T16:35:06/mapped_metadata_2024-04-19T16:57:54/data/1700.jsonl
[2024-04-19, 17:11:05 UTC] {{taskinstance.py:1345}} INFO - Marking task as FAILED. dag_id=harvest_collection, task_id=content_harvesting.content_harvest, map_index=47, execution_date=20240419T163454, start_date=20240419T170901, end_date=20240419T171105
[2024-04-19, 17:11:05 UTC] {{logging_mixin.py:150}} INFO - Message sent to SNS with Message ID: 61459648-c039-5f37-9f61-c8d467d003bb
[2024-04-19, 17:11:05 UTC] {{standard_task_runner.py:104}} ERROR - Failed to execute job 76793 for task content_harvesting.content_harvest (This task is not in success state - last 100 logs from Cloudwatch:
Traceback (most recent call last):
  File "/content_harvester/by_record.py", line 150, in get_dimensions
    return Image.open(filepath).size
           ^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/PIL/Image.py", line 3339, in open
    raise UnidentifiedImageError(msg)
PIL.UnidentifiedImageError: cannot identify image file '/tmp/f98a6a0f011057e206ebecb30c5e3075.jpg'
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/content_harvester/by_page.py", line 98, in <module>
    print_value.append(harvest_page_content(
                       ^^^^^^^^^^^^^^^^^^^^^
  File "/content_harvester/by_page.py", line 31, in harvest_page_content
    record_with_content = harvest_record_content(
                          ^^^^^^^^^^^^^^^^^^^^^^^
  File "/content_harvester/by_record.py", line 112, in harvest_record_content
    dimensions = get_dimensions(thumbnail.tmp_filepath, record['calisphere-id'])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/content_harvester/by_record.py", line 152, in get_dimensions
    raise Exception(
Exception: PIL.UnidentifiedImageError for calisphere-id oai:history.santacruzpl.org:129307: cannot identify image file '/tmp/f98a6a0f011057e206ebecb30c5e3075.jpg'
```

**Mapped metadata:**
`{"calisphere-id": "oai:history.santacruzpl.org:129307", "collection_data": ["27143::Local News Index"], "collection_name": ["Local News Index"], "collection_url": ["27143"], "creator": ["McCord, Shanna"], "date": ["2013-12-03"], "description": ["Tannery Arts Center"], "fetcher_type": ["oai"], "format": ["Text"], "id": "7485d02319dba4924e66040e9072d35d", "identifier": ["CF-53518"], "is_shown_at": "https://history.santacruzpl.org/omeka/items/show/129307", "is_shown_by": "https://history.santacruzpl.org/omeka/files/thumbnails/f98a6a0f011057e206ebecb30c5e3075.jpg", "item_count": 0, "language": ["English"], "mapper_type": ["oai.omeka.nothumb"], "media_source": {}, "publisher": ["Santa Cruz Public Libraries"], "repository_data": ["245::Santa Cruz Public Libraries"], "repository_name": ["Santa Cruz Public Libraries"], "repository_url": ["245"], "rights": ["Please contact the contributing institution for more information regarding the copyright status of this object."], "sort_collection_data": ["local news index::Local News Index::27143"], "sort_title": "art of collaboration", "source": ["Santa Cruz Sentinel"], "subject": ["Tanneries", "Art and Artists"], "temporal": ["2010s"], "thumbnail_source": "https://history.santacruzpl.org/omeka/files/thumbnails/f98a6a0f011057e206ebecb30c5e3075.jpg", "title": ["The art of collaboration"], "type": ["text"], "url_item": "https://history.santacruzpl.org/omeka/items/show/129307"}`

## 27148 content harvest page 0
**Content harvest error log**
```
Harvesting content for 49 records at 27148/vernacular_metadata_2024-04-19T18:49:55/mapped_metadata_2024-04-19T18:50:53/data/1.jsonl
[2024-04-19, 18:56:07 UTC] {{taskinstance.py:1345}} INFO - Marking task as FAILED. dag_id=harvest_collection, task_id=content_harvesting.content_harvest, map_index=0, execution_date=20240419T184944, start_date=20240419T185403, end_date=20240419T185607
[2024-04-19, 18:56:07 UTC] {{logging_mixin.py:150}} INFO - Message sent to SNS with Message ID: 9c0456d2-f875-54cd-a9f4-084e38f2a104
[2024-04-19, 18:56:07 UTC] {{standard_task_runner.py:104}} ERROR - Failed to execute job 76958 for task content_harvesting.content_harvest (This task is not in success state - last 100 logs from Cloudwatch:
Traceback (most recent call last):
  File "/content_harvester/by_record.py", line 150, in get_dimensions
    return Image.open(filepath).size
           ^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/PIL/Image.py", line 3339, in open
    raise UnidentifiedImageError(msg)
PIL.UnidentifiedImageError: cannot identify image file '/tmp/a3b3a58e583a985aa6b65510210a8eb2.jpg'
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/content_harvester/by_page.py", line 98, in <module>
    print_value.append(harvest_page_content(
                       ^^^^^^^^^^^^^^^^^^^^^
  File "/content_harvester/by_page.py", line 31, in harvest_page_content
    record_with_content = harvest_record_content(
                          ^^^^^^^^^^^^^^^^^^^^^^^
  File "/content_harvester/by_record.py", line 112, in harvest_record_content
    dimensions = get_dimensions(thumbnail.tmp_filepath, record['calisphere-id'])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/content_harvester/by_record.py", line 152, in get_dimensions
    raise Exception(
Exception: PIL.UnidentifiedImageError for calisphere-id oai:history.santacruzpl.org:86: cannot identify image file '/tmp/a3b3a58e583a985aa6b65510210a8eb2.jpg'
```

**Mapped metadata**
`{"calisphere-id": "oai:history.santacruzpl.org:86", "collection_data": ["27148::Faye Ellis Collection"], "collection_name": ["Faye Ellis Collection"], "collection_url": ["27148"], "coverage": ["Ben Lomond"], "date": ["1885-08-20"], "fetcher_type": ["oai"], "format": ["TEXT"], "id": "c7397e2c99b5aa6df3e94a1919702fea", "identifier": ["FE-BEN-FIRES-02", "FOLDER-BEN_LOMOND-FIRES"], "is_shown_at": "https://history.santacruzpl.org/omeka/items/show/86", "is_shown_by": "https://history.santacruzpl.org/omeka/files/thumbnails/a3b3a58e583a985aa6b65510210a8eb2.jpg", "item_count": 0, "language": ["English"], "mapper_type": ["oai.omeka.nothumb"], "media_source": {}, "publisher": ["Santa Cruz Public Libraries"], "repository_data": ["245::Santa Cruz Public Libraries"], "repository_name": ["Santa Cruz Public Libraries"], "repository_url": ["245"], "rights": ["Please contact the contributing institution for more information regarding the copyright status of this object."], "sort_collection_data": ["faye ellis collection::Faye Ellis Collection::27148"], "sort_title": "ben lomond burned over the san vicente grant and adjoining farms devastated by fire", "source": ["Santa Cruz Surf"], "spatial": ["Ben Lomond"], "subject": ["Forest Fires", "Fire and Fire Protection"], "temporal": ["1880s"], "thumbnail_source": "https://history.santacruzpl.org/omeka/files/thumbnails/a3b3a58e583a985aa6b65510210a8eb2.jpg", "title": ["Ben Lomond Burned Over: The San Vicente Grant and Adjoining Farms Devastated by Fire"], "type": ["text"], "url_item": "https://history.santacruzpl.org/omeka/items/show/86"}`

## 27148 content harvest page 1
**content harvest error log**
```
Harvesting content for 50 records at 27148/vernacular_metadata_2024-04-19T18:49:55/mapped_metadata_2024-04-19T18:50:53/data/2.jsonl
PIL.UnidentifiedImageError: cannot identify image file '/tmp/d15d2283f6b2b3e7b7304fbe95bc30d0.jpg'
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/content_harvester/by_page.py", line 98, in <module>
    print_value.append(harvest_page_content(
                       ^^^^^^^^^^^^^^^^^^^^^
  File "/content_harvester/by_page.py", line 31, in harvest_page_content
    record_with_content = harvest_record_content(
                          ^^^^^^^^^^^^^^^^^^^^^^^
  File "/content_harvester/by_record.py", line 112, in harvest_record_content
    dimensions = get_dimensions(thumbnail.tmp_filepath, record['calisphere-id'])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/content_harvester/by_record.py", line 152, in get_dimensions
    raise Exception(
Exception: PIL.UnidentifiedImageError for calisphere-id oai:history.santacruzpl.org:109: cannot identify image file '/tmp/d15d2283f6b2b3e7b7304fbe95bc30d0.jpg'; 22966)
```

**Mapped Metadata**
`{"calisphere-id": "oai:history.santacruzpl.org:109", "collection_data": ["27148::Faye Ellis Collection"], "collection_name": ["Faye Ellis Collection"], "collection_url": ["27148"], "coverage": ["Ben Lomond"], "creator": ["Kelly, Mabel Hall [?]"], "description": ["\"This page copied from Mabel Hall Kelly material, Oct. 1978\" [Faye Ellis note]."], "fetcher_type": ["oai"], "format": ["TEXT"], "id": "4539ea9942b6e05b94463183013af184", "identifier": ["FE-BEN-TELEGRAPH_TELEPHONE-04", "FOLDER-BEN_LOMOND-TELEGRAPH_TELEPHONE"], "is_shown_at": "https://history.santacruzpl.org/omeka/items/show/109", "is_shown_by": "https://history.santacruzpl.org/omeka/files/thumbnails/d15d2283f6b2b3e7b7304fbe95bc30d0.jpg", "item_count": 0, "language": ["English"], "mapper_type": ["oai.omeka.nothumb"], "media_source": {}, "publisher": ["Santa Cruz Public Libraries"], "repository_data": ["245::Santa Cruz Public Libraries"], "repository_name": ["Santa Cruz Public Libraries"], "repository_url": ["245"], "rights": ["Please contact the contributing institution for more information regarding the copyright status of this object."], "sort_collection_data": ["faye ellis collection::Faye Ellis Collection::27148"], "sort_title": "ben lomond exchange", "spatial": ["Ben Lomond"], "subject": ["Telephone", "Ben Lomond Exchange", "Beede, Elizabeth"], "temporal": ["1920s", "1940s"], "thumbnail_source": "https://history.santacruzpl.org/omeka/files/thumbnails/d15d2283f6b2b3e7b7304fbe95bc30d0.jpg", "title": ["[The Ben Lomond Exchange]"], "type": ["text"], "url_item": "https://history.santacruzpl.org/omeka/items/show/109"}, `

**Comment by christinklez** [comment]
We ETL'd these two collections to Rikolti. But post-mvp, we should work with Santa Cruz Public Libraries to address these errors.

**Comment by christinklez** [comment]
Fixed via #899!

**Comment by christinklez** [comment]
To do:

- [ ] Harvest 27143 via omeka.nothumb
   - [ ] verify on -stage: https://calisphere-stage.cdlib.org/item/058690c76101663e538e2b47938d2e51/
   - [ ] verify on -staget: https://calisphere-stage.cdlib.org/item/7485d02319dba4924e66040e9072d35d/
- [ ] Harvest 27148 via omeka.nothumb
   - [ ] verify on -stage: https://calisphere-stage.cdlib.org/item/c7397e2c99b5aa6df3e94a1919702fea/
   - [ ] verify on -stage: https://calisphere-stage.cdlib.org/item/4539ea9942b6e05b94463183013af184/
